source/opt/fold_spec_constant_op_and_composite_pass.h

```diff
@@ -35,22 +35,13 @@ class FoldSpecConstantOpAndCompositePass : public Pass {
   FoldSpecConstantOpAndCompositePass();
 
   const char* name() const override { return "fold-spec-const-op-composite"; }
-  Status Process(ir::Module* module) override {
-    Initialize(module);
-    return ProcessImpl(module);
-  };
+
+  Status Process(ir::Module* module) override;
```

Comment:
Suspect you don't need algorithm or memory anymore.  Add it to the corresponding .cpp file?